dockstore-webservice/src/main/java/io/dockstore/webservice/languages/WDLHandler.java

```diff
@@ -427,10 +427,13 @@ public VersionTypeValidation validateTestParameterSet(Set<SourceFile> sourceFile
             // Iterate over each call, determine dependencies
             Map<String, List<String>> callsToDependencies = wdlBridge.getCallsToDependencies(tempMainDescriptor.getAbsolutePath(), mainDescName);
             toolInfoMap = mapConverterToToolInfo(callsToDockerMap, callsToDependencies);
+
             // Get import files
             namespaceToPath = wdlBridge.getImportMap(tempMainDescriptor.getAbsolutePath(), mainDescName);
-        } catch (IOException | NoSuchElementException | WdlParser.SyntaxError e) {
-            throw new CustomWebApplicationException("could not process wdl into DAG: " + e.getMessage(), HttpStatus.SC_INTERNAL_SERVER_ERROR);
+        } catch (WdlParser.SyntaxError e) {
+            return Optional.empty();
```

Comment:
As you suggested in your PR description, I think this should be a 400 error.

Reply:
Should CWLHandler throw a 400 as well when encountering parsing errors (such as invalid YAML syntax)?

Reply:
The initial commit in the PR handles the errors this way (both CWL and WDL), see [here](https://github.com/dockstore/dockstore/pull/3922/commits/8d19fddbf64385c8eec6ff456972d3ac3130de1d). I can revert back if that seems more appropriate.

Reply:
Not sure about the CWL case; in this case it definitely seems like it should be a 400 -- the method is not returning a useful response; it's not Dockstore's fault -- it's a WDL with syntax errors submitted by a user.

Reply:
[Fixed.](https://github.com/dockstore/dockstore/blob/c5277c33e2a3bf271f3be4e33b742fb646b1a076/dockstore-webservice/src/main/java/io/dockstore/webservice/languages/WDLHandler.java#L435)